### ISSUE TITLE
[Feature] Allow both, parallel and ordered test execution at the same time

### ISSUE THREAD

**Issue by dik-garri** [microsoft/playwright#17553]
As a playwright user, I want to be able to run some spec files in parallel and some in a specific order at the same time. 
I have a situation when I have a lot of spec files and some of them depend on others. 
Currently, I turned off parallel execution `workers: 1` but it increases time on all tests execution a lot

**Comment by dik-garri** [comment]
Any updates on this issue?

**Comment by Kamal-Yalikun** [comment]
Are there any updates on this issue? This is a very important feature for our purposes

**Comment by eatonjl** [comment]
I'm hoping for this functionality as well.

**Comment by younesjallouli** [comment]
Any update on this one?

**Comment by Terence625** [comment]
Upvote to this feature, and wonder why PR https://github.com/microsoft/playwright/pull/17548 is removed

**Comment by younesjallouli** [comment]
Same here, we really need this one.

Today there is no way to tell playwright, run these test files sequentially, but inside each test file, go full steam and run tests in parallel.

![image](https://github.com/microsoft/playwright/assets/116430697/cf03b480-6ddf-4eae-9356-0a91708b9628)

**Comment by FrankNutreco** [comment]
@younesjallouli  Have you tried the following?

In the playwright.config.ts file you can set the dependencies:
![image](https://github.com/microsoft/playwright/assets/107055245/83c4c232-6d07-474a-b4a6-f9d9a86834cb)

In the package.json file you can set the following:
![image](https://github.com/microsoft/playwright/assets/107055245/064771a5-ec06-413b-8829-23e74d51a6bc)

When running your tests, you can set the workers:
npm run tests:exampleTests -- --workers: 4



These test files will be run in sequential order:
setup, 
CreateExampleA,
CreateExampleB,
CreateExampleC

These test files will be run in parallel:
UpdateExampleA, UpdateExampleB

Inside each test file you can set .parallel to run testcases within a test file in parallel.

Downside to this method is when you have a file that has to be ran in sequential order and also contains testcases that cannot be run in parallel, since it will only use one worker and slows down the total run speed.

**Comment by younesjallouli** [comment]
I don't think this is the solution I was hoping for.
To give context: to test our SaaS app, we have multiple tests across multiple files, and we run these tests in parallel on multiple workers. However, you need to create some objects in previous tests, before moving on to the next tests (like testing registering a payment method, before testing the creation of a payment, before testing a refund of that payment). We do not believe in loading data in the DB and maintaining those scripts.

In an ideal world, there should be a mode in Playwright that tells Playwright to run files sequentially (create payment method first, then create payments, then ...) without even having to specify the test file names in the config files. Even if workers become available playwright should not pick up tests from the following file until the current file is 100% done.

**Comment by Terence625** [comment]
> I don't think this is the solution I was hoping for. To give context: to test our SaaS app, we have multiple tests across multiple files, and we run these tests in parallel on multiple workers. However, you need to create some objects in previous tests, before moving on to the next tests (like testing registering a payment method, before testing the creation of a payment, before testing a refund of that payment). We do not believe in loading data in the DB and maintaining those scripts.
> 
> In an ideal world, there should be a mode in Playwright that tells Playwright to run files sequentially (create payment method first, then create payments, then ...) without even having to specify the test file names in the config files. Even if workers become available playwright should not pick up tests from the following file until the current file is 100% done.

I believe I have the same situation as you here: https://github.com/microsoft/playwright/issues/29282. I was using serial mode, but it's not recommended, maybe you can try this solution https://github.com/microsoft/playwright/issues/29282#issuecomment-1920505421, I'm using project dependencies for multiple applications.

**Comment by younesjallouli** [comment]
Any news on @dik-garri 's request playwright team? @yury-s @mxschmitt @pavelfeldman 

This feature is a no-brainer for me. It was asked many times by the community

**Comment by zargham-leanix** [comment]
@mxschmitt  any update on this ?

**Comment by michaeljoshuaclaro** [comment]
> I don't think this is the solution I was hoping for. To give context: to test our SaaS app, we have multiple tests across multiple files, and we run these tests in parallel on multiple workers. However, you need to create some objects in previous tests, before moving on to the next tests (like testing registering a payment method, before testing the creation of a payment, before testing a refund of that payment). We do not believe in loading data in the DB and maintaining those scripts.
> 
> In an ideal world, there should be a mode in Playwright that tells Playwright to run files sequentially (create payment method first, then create payments, then ...) without even having to specify the test file names in the config files. Even if workers become available playwright should not pick up tests from the following file until the current file is 100% done.

I agree, I wish playwright would have that feature wherein we can prioritize a test first before running another set of tests.

**Comment by elyamurza** [comment]
+1, feature would be helpful.

**Comment by Estegar** [comment]
+1

**Comment by akash-k-cs** [comment]
+1

**Comment by younesjallouli** [comment]
@yury-s @mxschmitt @pavelfeldman Please some update :D

**Comment by hekin1** [comment]
+1

**Comment by ivansansao** [comment]
Hi everyone,

First of all, congratulations for the Playwrigth project.

I have the same issue here, I am forced to use serial for all because I can't do a parallel block inside a serial block.

Tks in advanced!

**Comment by ivansansao** [comment]
**Hi everyone,**

For now, I'm using this crazy workaround (horrible, very painful, not scalable - you lose UI control, and so on)

It is something like:

**`run-tests.sh`**:

```bash
npx playwright test tests/serial-things1 --workers=1
npx playwright test tests/parallel-thing1 --workers=100%
npx playwright test tests/serial-things2 --workers=1
npx playwright test tests/parallel-thing2 --workers=100%
npx playwright test tests/serial-things3 --workers=1
npx playwright test tests/parallel-thing3 --workers=100%
```
